Sources/SystemInternals/Exports.swift

```diff
@@ -36,47 +50,129 @@ public var system_errno: CInt {
 #endif
 
 public func system_open(_ path: UnsafePointer<CChar>, _ oflag: Int32) -> CInt {
+#if os(Windows)
+  var fh: CInt = -1
+  _ = _sopen_s(&fh, path, oflag, _SH_DENYNO, _S_IREAD | _S_IWRITE)
```

Comment:
Is this the rough equivalent of POSIX open? E.g., what is the difference between a Windows file handle and a POSIX file descriptor?

Reply:
Yes, this is roughly the equivalent of the POSIX `open`.

I don't think that its valuable to try to enumerate every single difference.  What type of differences are you really looking for?

But for a taste, there are a large number of differences between a file handle and a POSIX file.  The POSIX file is a user mode only, process specific construct.  It _cannot_ be shared across processes.  It is a mapping from the kernel side representation (`HANDLE`) and an application specific view.  It is maintained by means of a lookup table in the application's address space by the C library.  Additionally, it has a comparatively restricted set of operations available to it.